### DISCUSSION TITLE
Recomended firmware for TUBE products.

### DISCUSSION THREAD

**Discussion by MattWestb** [tube0013/tube_gateways#93] [Ideas]
### First thanks for the Discussion forum you have opened !!!

I was adding EZSP 6.10.x for some devices as candidate for recommended stable firmware in the Zigpy Wiki and all your is one the stable 6.7.X and have no linked 6.10 candidates (I finding 8 and earlier 9 is not candidates for my but perhaps for you) and some new products is missing in Zigpy Wiki.

My question is do you planing cooking 6.10.x for not current products and if yes pleas adding it in the Zigpy wiki then you have getting it in place.

I was thinking doing one matrix for RCP and NCP / RCP firmware but we have not so many to testing for the moment (only our Sky Connect for all and My Billy for RPC and some TUBEs) im curious if you have some in the pocket for the future.

Looks like MG1P is getting RCP updates but not MG1B (tuya and other) but i have not time testing the latest updated GSDK but i hope all MG2 and MG12/13 devices is coming (also IKEA Marcus / Silabs MG21 light module).

The EM35X is 110% not getting any RCP (the old tool change is not supported by OT) but i think most have being great OTRB / Zigbee coordinator with the update but dead is dead and user **shall** baying IKEA devices and destroying them and making Billy or Marcus NCPs / RCPs of then and then its not working they is going TUBE for GOLD !!!

**Comment by tube0013** [comment]
Hi @MattWestb,

EZSP FW wise. I have internal 6.10.3 build for the MGM12 that I'm running my self. and I plan to get some up for the MGM212. I've tried to get 7.1.1 working, but have not gotten a good build yet. It will work but ezsp crashes and restarts a few times a day (ran it for a month or so before reverting).

I will definitely have RCP FW. hopefully both variants. but right now the rcp+ncp variant seems to like HW Flow which I don't / can't have with the network connected serial.

I have a stash of MGM210 modules I'm going to start rolling out into ethernet/usb and Poe models soon. Now that I've seem to finally have a stable serial on current esphome! https://github.com/tube0013/esphome-stream-server-v2

they've been a long time coming. https://twitter.com/TubesZb/status/1525642461367808002

**Reply by MattWestb** [reply]
For my is NCP 7.X still very early Alpha and all things is not in place and hope going to normal beta for debugging and fixing. Then is in late beta or GA its being interesting and getting new good functions on or MG2X and MG1XX devices (RCP is "other thing") and MG1P (Billy) Is NCP starting (allocating NVM3 blocks) but the com is not working then luck of working SW flow control and the support is gone in the last GSDK update.

### You have getting steam server with sensor working !!!!!
I must trying it on some D1 Mini that is frozen on the last working one !!!


On more interesting thing with `Zeroconf` in HA.
I was thinking cloning yours for Billys but its not looking nice.

Shall we making one for `NPC`(EZSP), `RCP`(OTBR), `RPC / RPC` (Zigbeed and OTBR) and  `NCP / RPC` (EZSP and OTBR) with 115200 and SW without branding so its "open" for all network attached also Tasmota if they is using the same settings in HA ?
Perhaps ZHA can extracting manufacture and board ID tokens from the flash then its looks very nice for all that have making OK (My Sky Connect was wrongly made but i hope being fixed in GA).

The bad thing Silabs have blocking TI and other in the last version then changing the protocol but have open for NCP and RCP at the same time.

**Reply by MattWestb** [reply]
Have you trying adding the hardware pin definition in ESPHome UART config and using HW Flow control in both  ends ?
If its using 100% native raw stream and the UART is doing it also with the HW defined it shall working (I think = not knowing).

**Reply by tube0013** [reply]
Hardware UART in the ESP is different than Hardware Flow. As far as I know hardware flow is not supported in the esphome uart implementation - https://esphome.io/components/uart.html

**Reply by tube0013** [reply]
maybe it will come at some point, but lots of hurdles. this is using IDF, current esphome does not support ethernet when using IDF framework so falls back to Auduino.
